qiime2/sdk/proxy.py

```diff
@@ -0,0 +1,193 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2023, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+from qiime2.core.type.util import is_visualization_type, is_collection_type
+
+
+class Proxy:
```

Comment:
We may need to implement things like `__eq__` and friends from the base class, since we would want to block.